lib/components/data/plots/CWMSPlot.jsx

```diff
@@ -94,86 +145,69 @@ export default function CWMSPlot({
-
-    // Force the font size if specified
-    if (plotFontSize) layout["font"] = { size: plotFontSize };
-    else if (unit_keys.length > 4) {
-      layout["font"] = { size: 8 };
-    }
+    let ts_keys = Object.keys(tsData.ts);
 
-    // Create traces keyed to the unit
+    // Create traces keyed to the tsdata
+    let traces = [];
     let ts;
-    for (let k_idx = 0; k_idx < unit_keys.length; k_idx++) {
-      const key = unit_keys[k_idx];
-      for (let ts_idx = 0; ts_idx < tsData.ts[key].length; ts_idx++) {
-        ts = tsData.ts[key][ts_idx];
-        const trace = {
-          x: ts.values.map((value) => new Date(value[0])),
-          y: ts.values.map((value) => value[1]),
-          yaxis: "y" + trace_cnt,
-          type: "scatter",
-          mode: "lines",
-          showlegend: true,
-          legend: { x: 1, xanchor: "right", y: 1 },
-          name: `${ts.name.split(".")[1]}<br>${ts.name.split(".")[5]}`,
-        };
-        traces.push(trace);
-      }
-      let title_text = `${ts.name.split(".")[1]}<br>(${key})`;
-      if (unit_keys.length > 3) title_text = key;
-
-      layout["yaxis" + trace_cnt] = {
-        domain: [
-          Math.round(domain_start * 100) / 100,
-          Math.round(domain_end * 100) / 100,
-        ],
-        title: { text: title_text },
-      };
-      // Alter if there are more than 4 units / view ports
-      if (unit_keys.length > 4) {
-        layout["yaxis" + trace_cnt]["nticks"] = 1;
-        layout["yaxis" + trace_cnt]["tickvals"] = [
-          ts.values[ts.values.length / 2],
-        ];
-        layout["yaxis" + trace_cnt]["dtick"] = 1;
-        layout["yaxis" + trace_cnt]["tick0"] = 0.5;
+
+    // Loop thru TS Data for timeseries and location levels
+    for (let k_idx = 0; k_idx < ts_keys.length; k_idx++) {
+      const key = ts_keys[k_idx];
+
+      // Add Timeseries to list of traces
+      if (tsids.includes(key)) {
+        for (let ts_idx = 0; ts_idx < tsData.ts[key].length; ts_idx++) {
+          ts = tsData.ts[key][ts_idx];
+          const params = timeseriesParams.filter((item) => item.tsid == ts.name)
+          const trace = {
+            x: ts.values.map((value) => new Date(value[0])),
+            y: ts.values.map((value) => value[1]),
+            name: params[0].name,
+            mode: params[0].mode,
+            type: params[0].type,
+            marker: params[0].marker,
+            line: params[0].line,
+            showlegend: true,
+            legend: { x: 1, xanchor: "right", y: 1 },
+            yaxis: params[0].yaxis,
+          }
+          traces.push(trace)
+        }
       }
 
-      domain_start += domain_delta;
-      domain_end += domain_delta;
-      trace_cnt++;
+      // Add Location Levels to list of traces
+      if (levels.includes(key)) {
+        for (let ts_idx = 0; ts_idx < tsData.ts[key].length; ts_idx++) {
+          ts = tsData.ts[key][ts_idx];
+          const params = locationLevelParams.filter((item) => item.levelid == key)
```

Comment:
This is pedantic, but you might consider throwing a `[0]` on the end here so you don't have to keep specifying it below.